packages/@glimmer/runtime/lib/vm/rehydrate-builder.ts

```diff
@@ -379,15 +379,21 @@ export class RehydrateBuilder extends NewElementBuilder implements ElementBuilde
   __pushRemoteElement(
     element: SimpleElement,
     cursorId: string,
-    nextSibling: Option<SimpleNode> = null
+    insertBefore: Option<null>
   ): Option<RemoteLiveBlock> {
     let marker = this.getMarker(element as HTMLElement, cursorId);
 
     if (marker.parentNode === element) {
+      if (insertBefore !== null) {
```

Comment:
It's my understanding that we want `null` to mean append, `undefined` to mean replace, and any other value to mean "before this thing" (though this option isn't implemented yet). If thats the case, the current logic here is going to make fixing that third scenario pretty hard. How about changing the logic of the conditional to something like:

```suggestion
      if (insertBefore === undefined) {
```